src/parse.c

```diff
@@ -9398,6 +9478,7 @@ static void p_selected_waveforms(tree_t stmt, tree_t target, tree_t reject)
          tree_set_value(tree_assoc(stmt, i), a);
 
       tree_set_loc(a, CURRENT_LOC);
+      tree_set_ident(a, get_implicit_label(a, CURRENT_LOC));
```

Comment:
Shouldn't these all be `_S0` as they will be the first statement inside the process?

Reply:
IMHO no. I tried:

`with clk_sys select clk_n <=
        '0' when '1',
        '1' when others;`

which leads to something like:

`P3: process is
    begin
      _S0: case CLK_SYS is
        when '1' =>
          _S1: CLK_N <= reject 0 ps inertial '0';
        when others =>
          _S2: CLK_N <= reject 0 ps inertial '1';
      end case;
      wait on CLK_SYS;   -- static
    end process;`

In that case, there is one _SX for case statement, and one for each signal assign for each alternative.
This is the same as when case statement is parsed.